internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -381,6 +381,52 @@ func isUserManagedNetwork(clusterInstall *hiveext.AgentClusterInstall) bool {
 	return false
 }
 
+func hyperthreadingInSpec(clusterInstall *hiveext.AgentClusterInstall) bool {
+	//see https://docs.openshift.com/container-platform/4.7/installing/installing_platform_agnostic/installing-platform-agnostic.html#installation-bare-metal-config-yaml_installing-platform-agnostic
+	return len(clusterInstall.Spec.Compute) > 0 || clusterInstall.Spec.ControlPlane != nil
+}
+
+func getHyperthreading(clusterInstall *hiveext.AgentClusterInstall) *string {
+	const (
+		None    = 0
+		Workers = 1
+		Masters = 2
+		All     = 3
+	)
+	var config uint = 0
+
+	//if there is no configuration of hyperthreading in the Spec then
+	//we are opting of the default behavior which is all enabled
+	if !hyperthreadingInSpec(clusterInstall) {
+		config = All
+	}
+
+	//check if the Spec enables hyperthreading for workers
+	if len(clusterInstall.Spec.Compute) > 0 {
+		if clusterInstall.Spec.Compute[0].Hyperthreading == "Enabled" {
```

Comment:
We need to update the spec to include a struct with name and boolean.
We will need to validate that the name can only be `worker` for now

Reply:
I will add it in a separate PR to avoid mixing API changes and Implementation